### ISSUE TITLE
Yolov8 obb prediction box rotation angle is very small

### ISSUE THREAD

**Issue by Joazs** [ultralytics/ultralytics#19246]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

I trained the Yolov8 obb model and found that the rotation angle was very small when making predictions. What could be the reason for this?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Joazs, thank you for bringing this to our attention and for your interest in Ultralytics 🚀! We recommend starting with the [Docs](https://docs.ultralytics.com/) for general usage and exploring resources like [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) examples, which often address many common questions.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us investigate the issue further.

If this is a ❓ Question about custom training or results, it would be great if you could share additional details such as your dataset configuration, training parameters, and any relevant logs or output. You can also review our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/) to rule out potential issues.

### Recommendations

Ensure you are using the latest version of the `ultralytics` package. Upgrade with the following command:

```bash
pip install -U ultralytics
```

### Environments

YOLO can be executed in the following verified environments, complete with essential dependencies like [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/), and [PyTorch](https://pytorch.org/) preinstalled:

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud**: Refer to the [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon Web Services**: See the [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**: See the [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

### Collaboration and Community

You can also connect with our awesome Ultralytics community for further discussions, support, and tips:
- [Discord](https://discord.com/invite/ultralytics) 🎧 for live chat
- [Discourse](https://community.ultralytics.com/) for detailed discussions
- [Subreddit](https://reddit.com/r/Ultralytics) for knowledge-sharing and fun discussions

### Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>  
Our CI tests ensure that all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) are verified daily and on every code update to maintain reliability on macOS, Windows, and Ubuntu.

This is an automated response to assist you promptly 😊. An Ultralytics engineer will also review and assist you with your query as soon as possible. Thank you for your patience and understanding!

**Comment by glenn-jocher** [comment]
@Joazs for OBB angle predictions in YOLOv8, small rotation values may stem from dataset characteristics (limited angle diversity) or angle parameterization in training. Ensure your dataset includes varied object orientations and verify angle annotations are correct. The model predicts rotation relative to object appearance in training data. For details on OBB formats and training, see our [OBB Task Documentation](https://docs.ultralytics.com/tasks/obb/).
